test-app/src/main/java/org/readium/r2/testapp/domain/Bookshelf.kt

```diff
@@ -110,19 +112,21 @@ class Bookshelf(
 
     private suspend fun addBookFeedback(
         url: AbsoluteUrl,
+        format: Format? = null,
         coverUrl: AbsoluteUrl? = null
     ) {
-        addBook(url, coverUrl)
+        addBook(url, format, coverUrl)
             .onSuccess { channel.send(Event.ImportPublicationSuccess) }
             .onFailure { channel.send(Event.ImportPublicationError(it)) }
     }
 
     private suspend fun addBook(
         url: AbsoluteUrl,
+        format: Format? = null,
         coverUrl: AbsoluteUrl? = null
     ): Try<Unit, ImportError> {
         val asset =
-            assetRetriever.retrieve(url)
+            assetRetriever.retrieve(url, FormatHints(format?.mediaType))
```

Comment:
Why don't you use the `format` directly here?

Reply:
Because we don't always have it, and the `AssetRetriever` doesn't accept an optional format. If I remember correctly, we even considered removing its method taking a `Format`.